io.ossia.score.yml

```diff
@@ -0,0 +1,333 @@
+      - -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+      - -DBUILD_SHARED_LIBS=OFF
+      - -DENABLE_WS=1
+      - -DENABLE_DTLS=0
+      - -DENABLE_EXAMPLES=0
+      - -DENABLE_DOCS=0
+      - -DENABLE_OSCORE=0
+      - -DENABLE_SERVER_MODE=0
+      - -DMAX_LOGGING_LEVEL=0
+    sources:
+      - type: archive
+        url: https://github.com/obgm/libcoap/archive/refs/tags/v4.3.4a.tar.gz
+        sha256: 45f1aabbb5f710e841c91d65fc3f37c906d42e8fc44dd04979e767d3960a77cf
+
+  # ossia uses boost
+  - name: boost
+    buildsystem: simple
+    build-commands:
+      - mkdir -p "${FLATPAK_DEST}/include/"
+      - cp -rf boost "${FLATPAK_DEST}/include/"
+    sources:
+      - type: archive
+        url: https://archives.boost.io/release/1.85.0/source/boost_1_85_0.tar.bz2
+        sha256: 7009fe1faa1697476bdc7027703a2badb84e849b7b0baad5086b087b971f8617
+
+  # HAP video decoding
+  - name: snappy
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+      - -DBUILD_SHARED_LIBS=OFF
+      - -DSNAPPY_BUILD_TESTS=0
+      - -DSNAPPY_BUILD_BENCHMARKS=0
+    sources:
+      - type: archive
+        url: https://github.com/google/snappy/archive/refs/tags/1.2.1.tar.gz
+        sha256: 736aeb64d86566d2236ddffa2865ee5d7a82d26c9016b36218fcc27ea4f09f86
+
+  # ossia uses mDNS for multiple protocol discovery mechanisms
+  - name: avahi
+    builddir: true
+    config-opts:
+      - --with-distro=none
+      - --disable-qt3
+      - --disable-qt4
+      - --disable-qt5
+      - --disable-gtk
+      - --disable-gtk3
+      - --disable-glib
+      - --disable-gobject
+      - --disable-python
+      - --disable-mono
+      - --disable-monodoc
+      - --disable-libevent
+      - --disable-libdaemon
+      - --disable-dbm
+      - --disable-gdbm
+      - --disable-manpage
+
+    post-install:
+      - make -C avahi-common install
+      - make -C avahi-client install
+
+    sources:
+      - type: archive
+        url: https://github.com/avahi/avahi/releases/download/v0.8/avahi-0.8.tar.gz
+        sha256: 060309d7a333d38d951bc27598c677af1796934dbd98e1024e7ad8de798fedda
+
+  - name: ossia-score
+    buildsystem: cmake-ninja
+    builddir: true
+    no-make-install: true
+    sources:
+      - type: archive
+        url: https://github.com/ossia/score/releases/download/v3.2.3/ossia.score-3.2.3-src.tar.xz
+        sha256: 0
+    config-opts:
+      # Note: don't use RelWithDebInfo as ossia release builds explicitly expect Release build type.
+      - -DCMAKE_BUILD_TYPE=Release
+      - -Wno-dev
+      - -DSCORE_DEPLOYMENT_BUILD=1
+      - -DSCORE_STATIC_PLUGINS=1
+      - -DSCORE_FHS_BUILD=1
+      - -DCMAKE_UNITY_BUILD=1
+      - -DCMAKE_SKIP_RPATH=ON
+      - -DOSSIA_USE_SYSTEM_LIBRARIES=1
+      - -DSCORE_USE_SYSTEM_LIBRARIES=1
+      - -DSCORE_NO_QTCONF=1
+
+    post-install:
+     - cmake -DCMAKE_INSTALL_DO_STRIP=1 -DCOMPONENT=OssiaScore -P cmake_install.cmake
```

Comment:
the install target isn't run by default?

Reply:
I disabled it explicitly - the default install target installs a lot of unnecessary things for this kind of linux packaging (but necessary in other cases). installing the "OssiaScore" component installs just the necessary things for the package.

Reply:
basically all the header files for the development SDK of the app - I don't think it's suitable for the application flatpak to provide that right ?

Reply:
that's why `cleanup` exist.

Reply:
I'll change it but I really think that flatpak should provide a way to specifiy which component to install instead of forcing to install a few thousand files just to remove them afterwards - this is just wasting CI time for no good reason

Reply:
You mean like `make-install-args` ?

(works with Ninja too)

Reply:
right now the regular install target is run anyway so I don't see your point.

Reply:
> otoh if `install-rule` calls `cmake --install` instead of the actual build system then it would work

no it calls `ninja` or `make` depending on the case.